students/itsabhi/Lesson3/strformat_lab.py

```diff
@@ -0,0 +1,78 @@
+#! /usr/bin/env python3
+
+def convert():
+    my_tuple = ( 2, 123.4567, 10000, 12345.67)
+    str_first = "{:0>3d}".format(my_tuple[0])
+    str_second = "{:.2f}".format(my_tuple[1])
+    str_third = "{:.2e}".format(my_tuple[2])
+    str_fourth = "{:.2e}".format(my_tuple[3])
+    print("file"+str_first + ": " + str_second + "," + str_third + "," + str_fourth)
+
+def convert1():
+    my_tuple = ( 2, 123.4567, 10000, 12345.67)
+    str_first = "{:0>3d}".format(my_tuple[0])
+    str_second = "{:.2f}".format(my_tuple[1])
+    str_third = "{:.2e}".format(my_tuple[2])
+    str_fourth = "{:.2e}".format(my_tuple[3])
+
+    print("file{}:,{},{},{}".format(str_first,str_second,str_third,str_fourth))
+    print(f"file{str_first}:,{str_second},{str_third},{str_fourth}") #using f-strings
```

Comment:
you can simplify both convert functions by creating the strings inline:
`print("file_{:03d} : {:10.2f}, {:.2e}, {:.3g}".format(2, 123.4567, 10000, 12345.67))
or
`print("file_{:03d} : {:10.2f}, {:.2e}, {:.3g}".format(*my_tuple))`